app/assets/javascripts/models/attachment_options.js

```diff
@@ -0,0 +1,63 @@
+export default class AttachmentOptions {
+
+  constructor({ refreshCallback }) {
+    this.refreshTime = 3600000;
+    this.timestampVerifierId;
+    this.refreshCallback = refreshCallback;
+
+    this.getOptions();
+  }
+
+  clear() {
+    clearTimeout(this.timestampVerifierId);
+  }
+
+  buildProps(args) {
+    const {
+      name, progressElementId,
+      finishedElementId, attachinaryContainerId,
+      filesModel, options
+    } = args;
+    let files = filesModel;
+
+    if(files) {
+      files = files.map(function(d) { return d.file });
+    }
+    const extended_options = $.extend(options.attachinary, {
+      files_container_selector: '#' + attachinaryContainerId,
+      'files': files
+    });
+    const dataAttachinary = JSON.stringify(extended_options);
+    const dataFormData = JSON.stringify(options.html.data.form_data);
+    const dataUrl = options.html.data.url;
+    const multiple = (options.html.multiple) ? 'multiple' : '';
+    return {
+      dataAttachinary: dataAttachinary,
+      dataFormData: dataFormData,
+      dataUrl: dataUrl,
+      multiple: multiple
+    }
+  }
+
+  fetchOptions() {
+    const request_options = {
+      type: 'GET',
+      dataType: 'json',
+      url: '/attachments/signature'
+    }
+    return $.ajax(request_options)
+  }
+
+  getOptions() {
```

Comment:
I would call this method simply `fetch()` since the class name already has the word "options" on it, what do you think?